src/garage/tf/algos/dqn.py

```diff
@@ -240,9 +241,11 @@ def optimize_policy(self, itr, samples_data):
         # normalize pixel to range [0, 1] since the samples stored in the
         # replay buffer are of type uint8 and not normalized, for memory
         # optimization
-        observations = normalize_pixel_batch(self.env_spec, observations)
-        next_observations = normalize_pixel_batch(self.env_spec,
-                                                  next_observations)
+
+        if isinstance(self.env_spec.observation_space, akro.Image):
```

Comment:
DQN can only use Image observations, so your code should actually just be raising an exception when you get a non-Image observation space.

Reply:
About this, this would actually break `examples/dqn_cartpole`. The observations in that example are Boxes relating to the position of the cart and pole, not RGB pixels.